### PR TITLE
feature/freebsd-support-one: main freebsd adjustments

### PR DESCRIPTION
**Pull request by cosmicpsyop**
**FreeBSD 13 install and build README**
https://github.com/hoytech/strfry/pull/80/files

**submodule work**
uWebSockets
https://github.com/hoytech/uWebSockets/pull/4/files
golpe
https://github.com/hoytech/golpe/pull/1/files

RELATED: https://github.com/hoytech/strfry/issues/18

### CODE REVIEW

**Comment by hoytech** [comment]
Wow, thank you!! I'm going to get FreeBSD up in a VM ASAP and test it out.

File: src/onAppStartup.cpp [hunk 1]
```diff
@@ -58,9 +58,21 @@ static void setRLimits() {
 
     if (getrlimit(RLIMIT_NOFILE, &curr)) throw herr("couldn't call getrlimit: ", strerror(errno));
```
Comment (cosmicpsyop) [inline src/onAppStartup.cpp:60]: on the system
```
~# sysctl kern.maxfiles kern.maxfilesperproc kern.openfiles 
kern.maxfiles: 129999
kern.maxfilesperproc: 116991
kern.openfiles: 112
~# ulimit -Hn
116991
~# ulimit -Sn
116991
~# ulimit -n
116991
~# ulimit -Hn
116991
```
refactor for testing to snarf as many handles available up to what was asked for

**Comment by cosmicpsyop** [comment]
double checked linux and freebsd such that both build clean.  outside the PR will revisit required packages for freebsd for readme,

**Comment by hoytech** [comment]
Sorry for the delay -- Looks good, merging. Thank you!

**Pull request merged**